### PR TITLE
chore(lib/babe): create BlockBuilder type

### LINKED ISSUE #882: separate block builder functionality from BABE session
## Expected Behavior

- the block building functionality (ie lib/babe/build.go) should be separated into a separate submodule
- this will make testing easier for other packages that need to build blocks, but don't need an entire BABE session

## Current Behavior

- block building requires a BABE session

## Possible Solution

- create a new BlockBuilder type that implements the functions in lib/babe/build.go

## Checklist

- [x] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by danigomez**
## Changes

- Allows to create a block using a BlockBuilder without the need of an entire BABE service instance.

## Tests

```

```

## Issues

- Closes #882

### CODE REVIEW

File: lib/babe/build.go [hunk 1]
```diff
@@ -18,25 +18,79 @@ package babe
...
+	rt                    runtime.Instance
+	keypair               *sr25519.Keypair
+	transactionState      TransactionState
+	blockState            BlockState
+	slotToProof           map[uint64]*VrfOutputAndProof
+	currentAuthorityIndex uint32
+}
+
+// nolint
+func NewBuilder(rt runtime.Instance, kp *sr25519.Keypair, ts TransactionState, bs BlockState, sp map[uint64]*VrfOutputAndProof, authidx uint32) (*BlockBuilder, error) {
```
Comment (EclesioMeloJunior) [inline lib/babe/build.go:67]: The name could be `NewBlockBuilder` as it is related to construct a `BlockBuilder`

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]

File: lib/babe/build.go (hunk 1 above)
Comment (danigomez) [reply lib/babe/build.go]: Done!

**Review by EclesioMeloJunior — APPROVED** [review]
lgtm

**Review by noot — APPROVED** [review]
looks good, thanks so much for the PR!

**Pull request merged**